### DISCUSSION TITLE
Listing WIQL Results Parent then Child

### DISCUSSION THREAD

**Discussion by lewginn** [rfennell/AzurePipelines#1427] [Q&A]
Hi, first off, this tool is great and I'm hugely thankful. 

Using the WIQL Query listing, I'm able to list two work item types from an Iteration that I want. 
They currently display in a long list as such:
- User Story1
- User Story2
- User Story3
- Task 1
- Task 2
- Task 3
- Task 4
- Task 5

See markdown below: 

```
## User Stories in this release: ({{queryWorkItems.length}})

##User Stories
{{#forEach queryWorkItems}}
#{{this.id}}
{{/forEach}}
```

Is it possible to present the list in a way that the Parent Item is shown, then each child record afterwards? For Example (assume tasks are children of their above User Stories.

- User Story 1
  - Task 1
- User Story 2
  - Task 2
  - Task 3
  - Task 4
- User Story 3
  - Task 5

TIA

**Comment by rfennell** [comment]
You are going to have to write a template that does filters and look ups

The logic will be 

```
foreach queryWorkItems
     if a User Story
          write out User Story WI
          look for all child task WI in the queryWorktems array (by ID)
               write out Child task WI
```

It will be something like this

```
{{#forEach this.queryWorkItems}}

{{#if (eq (get 'System.WorkItemType' this) 'User Story')}}
*  **{{this.id}}**  {{lookup this.fields 'System.Title'}}

{{#forEach this.relations}}
{{#if (contains this.attributes.name 'Child')}}
{{#with (lookup_a_work_item ../queryWorkItems this.url)}}
      - {{this.id}} - {{lookup this.fields 'System.Title'}}
{{/with}}
{{/if}}
{{/forEach}}

{{/if}}

{{/forEach}}
```
